pkg/ast/spl/spl.peg

```diff
@@ -951,6 +956,301 @@ TcOptionCMD <- option:("usenull" / "useother" / "nullstr" / "otherstr") {
+    }
+    
+    if num1 != nil {
+        coeff, err = strconv.ParseFloat(num1.(string), 64)
+        if err != nil || coeff < 1.0 || coeff >= base {
+            return nil, fmt.Errorf("Spl peg: Invalid coefficient in log span, err: %v", err)
+        }
+    }
+    
+    logSpan := &structs.LogSpan {
+        Coefficient: coeff,
+        Base: base,
+    }
+    spanOptions := &structs.BinSpanOptions {
+        LogSpan: logSpan,
+    }
+    binOptionArgs := &BinOptionArgs {
+        argOption: "span",
+        binOptions: &structs.BinCmdOptions {
+            BinSpanOptions: spanOptions,
+        },
+    }
+    return binOptionArgs, nil
+}
+/ "span" EQUAL spanLen:SpanLenOption {
+
+    spanOptions := &structs.BinSpanOptions {
+        BinSpanLength: spanLen.(*structs.BinSpanLength),
+    }
+
+    binOptionArgs := &BinOptionArgs {
+        argOption: "span",
+        binOptions: &structs.BinCmdOptions {
+            BinSpanOptions: spanOptions,
+        },
+    }
+    return binOptionArgs, nil
+} 
+
+BinCmdOption <- option:(BinOptionAlignTime / BinOptionMinSpan / BinOptionMaxBins / BinOptionStart / BinOptionEnd / BinOptionSpan) {
+    return option, nil
+}
+
+BinCmdOptionsList <- first: BinCmdOption rest: (SPACE BinCmdOption)* {
+    restSlice := rest.([]any)
+    mapOption := make(map[string]bool)
+    mapOption["minspan"] = false
+    mapOption["bins"] = false
+    mapOption["start"] = false
+    mapOption["end"] = false
+    mapOption["span"] = false
+    mapOption["aligntime"] = false
+
+    numOptions := 1 + len(restSlice)
+
+    binCmdOption := &structs.BinCmdOptions{}
+
+    for i := 0; i < numOptions; i++ {
+        var binOptArg *BinOptionArgs
+        if i == 0 {
+            binOptArg = first.(*BinOptionArgs)
+        } else {
+            separator := restSlice[i - 1].([]any)
+            binOptArg = separator[1].(*BinOptionArgs)
+        }
+
+        argOption := binOptArg.argOption
+        val, exist := mapOption[argOption]
+        if !exist {
+            return nil, fmt.Errorf("Not a Valid Bin Command Option: %v", argOption)
+        }
+        if val {
+            return nil, fmt.Errorf("Invalid Syntax, option: %v is already specified", argOption)
+        }
+        mapOption[argOption] = true
+
+        switch argOption {
+            case "minspan":
+                binCmdOption.MinSpan = binOptArg.binOptions.MinSpan
+            case "span":
+                binCmdOption.BinSpanOptions = binOptArg.binOptions.BinSpanOptions
+            case "bins":
+                binCmdOption.MaxBins = binOptArg.binOptions.MaxBins
+            case "start":
+                binCmdOption.Start = binOptArg.binOptions.Start
+            case "end":
+                binCmdOption.End = binOptArg.binOptions.End
+            case "aligntime":
+                binCmdOption.AlignTime = binOptArg.binOptions.AlignTime
+            default:
+                return nil, fmt.Errorf("Not a Valid Bin Command Option: %v", argOption)
+        }
```

Comment:
I don't think you need the `mapOption`; the `default` case in your `switch` should catch that issue. If you have it to detect if an option is specified twice, you could instead have a set (a `map[string]struct{}`) of options that were set; that way you don't have to have the setup on lines 1153-1158

Reply:
addressed in the latest commit.